packages/grafana-prometheus/src/querybuilder/shared/QueryEditorHints.tsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from 'react';
+
+import { QueryHint } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+import { Button, Tooltip } from '@grafana/ui';
+
+import { PromQueryEditorProps } from '../../components/types';
+
+export function QueryEditorHints(props: PromQueryEditorProps) {
+  const [hints, setHints] = useState<QueryHint[]>([]);
+
+  useEffect(() => {
+    const query = { expr: props.query.expr, refId: props.query.refId };
+    const hints = props.datasource.getQueryHints(query, props.data?.series || []).filter((hint) => hint.fix?.action);
+    setHints(hints);
+  }, [props.datasource, props.data, props.query]);
```

Comment:
How about extracting those in the above and use it here?

```ts
// i.e.
const {datasource, data, query} = props;
// use it in useEffect
``

Reply:
I have extracted the props before using them as you have suggested.